test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -2997,24 +3123,32 @@ TEST_F(WeightedRoundRobinTest, CallAndServerMetric) {
   const int kNumServers = 3;
   StartServers(kNumServers);
   // Report server metrics that should give 1:2:4 WRR picks.
-  servers_[0]->server_metric_recorder_->SetCpuUtilization(0.9);
-  servers_[0]->server_metric_recorder_->SetQps(9);
-  servers_[1]->server_metric_recorder_->SetCpuUtilization(0.3);
-  servers_[1]->server_metric_recorder_->SetQps(6);
+  // weights = qps / (cpu_util + (eps/qps)) =
+  //   1/(0.4+0.4) : 1/(0.2+0.2) : 2/(0.3+0.1) = 1:2:4
+  servers_[0]->server_metric_recorder_->SetCpuUtilization(0.4);
+  servers_[0]->server_metric_recorder_->SetEps(40);
+  servers_[0]->server_metric_recorder_->SetQps(100);
+  servers_[1]->server_metric_recorder_->SetCpuUtilization(0.2);
+  servers_[1]->server_metric_recorder_->SetEps(20);
+  servers_[1]->server_metric_recorder_->SetQps(100);
   servers_[2]->server_metric_recorder_->SetCpuUtilization(0.3);
-  servers_[2]->server_metric_recorder_->SetQps(12);
+  servers_[2]->server_metric_recorder_->SetEps(5);
+  servers_[2]->server_metric_recorder_->SetQps(200);
   // Create channel.
   auto response_generator = BuildResolverResponseGenerator();
   auto channel = BuildChannel("", response_generator);
   auto stub = BuildStub(channel);
   response_generator.SetNextResolution(GetServersPorts(),
                                        kServiceConfigPerCall);
-  // Send requests with per-call reported QPS set to 100.
-  // This should override per-server QPS above and give 1:3:3 WRR picks.
+  // Send requests with per-call reported EPS/QPS set to 0/100.
+  // This should override per-server QPS above and give 1/4:1/2:1/3 = 3:6:4 WRR
+  // picks.
   EchoRequest request;
+  request.mutable_param()->mutable_backend_metrics()->set_eps(
+      std::numeric_limits<double>::min());
```

Comment:
Confusingly, that's ::lowest(). ::min() returns the smallest normalized positive number.

Reply:
Due to proto3 optional/default weirdness, I commented. This is test-specific issue since we are simulating per-call report in the echo request. I could instead use a non-zero eps but found this makes getting the expected weight math right rather complicated, so this looks better to follow.